Administrator Guide/Split brain and ways to deal with it.md

```diff
@@ -0,0 +1,61 @@
+1. Due to network disconnect:
+Where a client temporarily loses connection to the bricks.
+2. Gluster brick processes going down or returning error:
+Where server1 goes out of service and writes happen on server2. Server1 comes back but server2 goes out of service without the files having been healed. Write occurs on server1 and when server2 is returned to service, each has writes independent of one another.
+
+If we use the replica 2 volume, then there are high chances of ending up in split-brain.
+
+### Ways to deal with split brain:
+In glusterfs there are ways to resolve split brain. You can see the detailed description of automatically resolving split-brain [here](https://gluster.readthedocs.io/en/latest/Troubleshooting/heal-info-and-split-brain-resolution/) and manual resolution [here](https://gluster.readthedocs.io/en/latest/Troubleshooting/split-brain/). Moreover, there are ways which can guard us from ending up in split brain situations. The ways which guards us against split brain are:
+1. Replica 3 volume
+2. Arbiter volume
+
+Both of these uses the client-quorum option of glusterfs, avoid the split-brain situations.
+
+### Client quorum:
+This is a feature implemented in AFR to prevent split-brains in the I/O path for replicate/distributed-replicate volumes. By default, if the client-quorum is not met for a particular replica subvol, it becomes read-only. The other subvols (in a dist-rep volume) will still have R/W access. [Here](https://gluster.readthedocs.io/en/latest/Administrator%20Guide/arbiter-volumes-and-quorum/#client-quorum) you can see more details about client-quorum.
+
+### Replica 2 volume:
+In a replica 2 volume it is not possible to achieve high availability and consistency at the same time, without sacrificing tolerance to partition. If we set the client-quorum option to auto, then the first brick must always be up, irrespective of the status of the second brick. If only the second brick is up, the subvolume becomes EROFS. If the quorum-type is set to fixed, then the quorum-count is 1, then we may end up in split brain. i.e., One write succeed on one brick and other write on another brick. To avoid this we have to set the quorum-count to 2, which will cost the availability. Even if we have one replica brick up and running, we will end up seeing EROFS.
+
+#### 1. Replica 3 volume:
+When we create a replicated or distributed replicated volume with replica count 3, the cluster.quorum-type option is set to auto by default. That means at least 2 bricks should be up and running to satisfy the quorum and allow the writes. This is the recommended setting for a replica 3 volume and this should not be changed. Here is how it prevent files from ending up in split brain:
+
+B1, B2, and B3 are the 3 bricks of a replica 3 volume.
+1. B1 & B2 are up and B3 is down. Quorum is met and write happens on B1 & B2. Pending xattrs for B3 is set on B1 & B2.

```

Comment:
Gluster devs know what a `pending xattr` is but the users may not as you are using it for the first time in this document. Is there a need for them to know what a pending xattr is ? I don't think so.